test/e2e/apimachinery/webhook.go

```diff
@@ -1201,7 +1202,8 @@ func testAttachingPodWebhook(f *framework.Framework) {
 	ginkgo.By("'kubectl attach' the pod, should be denied by the webhook")
 	timer := time.NewTimer(30 * time.Second)
 	defer timer.Stop()
-	_, err = framework.NewKubectlCommand(f.Namespace.Name, "attach", fmt.Sprintf("--namespace=%v", f.Namespace.Name), pod.Name, "-i", "-c=container1").WithTimeout(timer.C).Exec()
+	tk := e2ekubectl.NewTestKubeconfig(framework.TestContext.CertDir, framework.TestContext.Host, framework.TestContext.KubeConfig, framework.TestContext.KubeContext, framework.TestContext.KubectlPath, f.Namespace.Name)
+	_, err = e2ekubectl.NewKubectlCommand(tk, "attach", fmt.Sprintf("--namespace=%v", f.Namespace.Name), pod.Name, "-i", "-c=container1").WithTimeout(timer.C).Exec()
```

Comment:
just leaving this as a thought (i dont think we should block this PR on this comment) but it may be good to give some thought on the spec for e2ebuectl.
There is a good deal of repetition and tons of verbosity because of how data is retrieved from the framework's test context and some ambiguity on where and how it is actually needed.
Another thing is that these `tk` structs are passed as method arguments when it may be better to associate the methods to these structs directly.